projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js

```diff
@@ -81,8 +82,31 @@
 							'&_ajax_nonce=' +
 							jetpackAdminMenu.jitmDismissNonce
 					);
-				} );
-			}
+				}
+			} );
+
+			var xhr = new XMLHttpRequest();
+			xhr.open(
+				'GET',
+				ajaxurl + '?action=upsell_nudge_jitm&_ajax_nonce=' + jetpackAdminMenu.upsellNudgeJitm,
+				true
+			);
+			xhr.setRequestHeader( 'X-Requested-With', 'XMLHttpRequest' );
+			xhr.withCredentials = true;
+			xhr.onreadystatechange = function () {
+				try {
+					if ( xhr.readyState === XMLHttpRequest.DONE ) {
+						if ( xhr.status === 200 ) {
+							adminMenu
+								.querySelector( '#toplevel_page_site_card' )
+								.insertAdjacentHTML( 'afterend', xhr.responseText );
+						}
+					}
+				} catch ( error ) {
+					// On failure, we just won't display an upsell nudge
+				}
+			};
+			xhr.send();
```

Comment:
Minor and non-blocking: if you update the `makeAjaxRequest` function to accept an optional callback, you could remove some code here.

```diff
diff --git a/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js b/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js
index ce3859138d..261c4ab150 100644
--- a/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js
+++ b/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js
@@ -86,7 +86,7 @@
                }
        }
 
-       function makeAjaxRequest( method, url, contentType, body ) {
+       function makeAjaxRequest( method, url, contentType, body, callback ) {
                var xhr = new XMLHttpRequest();
                xhr.open( method, url, true );
                xhr.setRequestHeader( 'X-Requested-With', 'XMLHttpRequest' );
@@ -94,6 +94,11 @@
                        xhr.setRequestHeader( 'Content-Type', contentType );
                }
                xhr.withCredentials = true;
+               if ( callback ) {
+                       xhr.onreadystatechange = function() {
+                               callback( xhr );
+                       }
+               }
                xhr.send( body );
        }
```

```suggestion
			makeAjaxRequest(
				'GET',
				ajaxurl + '?action=upsell_nudge_jitm&_ajax_nonce=' + jetpackAdminMenu.upsellNudgeJitm,
				null,
				null,
				function ( xhr ) {
					try {
						if ( xhr.readyState === XMLHttpRequest.DONE ) {
							if ( xhr.status === 200 ) {
								adminMenu
									.querySelector( '#toplevel_page_site_card' )
									.insertAdjacentHTML( 'afterend', xhr.responseText );
							}
						}
					} catch ( error ) {
						// On failure, we just won't display an upsell nudge
					}
				};
			);
```

Reply:
Addressed in https://github.com/Automattic/jetpack/pull/24304/commits/3a64f82b16e2aa793fb1976c4b9f806ad5f8847b. LoC-wise, things aren't much different, but at least we're reusing an existing function 😉